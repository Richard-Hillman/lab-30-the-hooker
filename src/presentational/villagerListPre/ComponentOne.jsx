/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import Villager from '../villagerPre/ComponentTwo';
import { Link } from 'react-router-dom';
// import styles from './ComponentOne.css';

const VillagersList = ({ villagers }) => {
  const villagerElements = villagers.map((villager) => (
    <Link key={villager._id} to = {`/containerTwo/${villager._id}`} >
      <li key={villager.name}>
        <Villager
          image={villager.image}
          name={villager.name}
          quote={villager.quote}
        />
      </li>
    </Link> 
  ));

  return ( 
    <ul data-testId="villagers"> 
      {villagerElements}
    </ul>
  );

};

VillagersList.propTypes = {
  villagers: PropTypes.array.isRequired,
};

export default VillagersList;
