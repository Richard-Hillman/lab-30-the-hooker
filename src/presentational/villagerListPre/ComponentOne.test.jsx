import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ComponentOne from './ComponentOne';

describe('ComponentOne component', () => {
  afterEach(() => cleanup());
  it('renders ComponentOne', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ComponentOne />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
