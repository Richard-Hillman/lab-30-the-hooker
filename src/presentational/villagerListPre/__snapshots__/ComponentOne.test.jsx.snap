// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentOne component renders  1`] = `
<DocumentFragment>
  <ul
    data-testid="villagers"
  />
</DocumentFragment>
`;
